cmd/agent/main.go

```diff
@@ -209,121 +209,122 @@ func run(ctx context.Context) error {
 		return err
 	}
 
-	// Check if secure connection can be made successfully
-	var httpClient = &http.Client{
-		Timeout: time.Second * 5,
-	}
+	topContext = context.WithValue(topContext, cavalidator.CacertsValid, false)
 
 	// Perform root CA verification
 	var transport *http.Transport
-	var rootCAFound bool
+	systemStoreConnectionCheckRequired := true
 	transport = rootCATransport()
 	if transport != nil {
 		logrus.Infof("Testing connection to %s using trusted certificate authorities within: %s", server, caFileLocation)
-		httpClient.Transport = transport
-		rootCAFound = true
-	} else if transport == nil && cluster.CAStrictVerify() {
-		logrus.Errorf("Strict CA verification is enabled but encountered error finding root CA: %v", err)
+		var httpClient = &http.Client{
+			Timeout:   time.Second * 5,
+			Transport: transport,
+		}
+		if _, err = httpClient.Get(server); err != nil && cluster.CAStrictVerify() {
+			logrus.Errorf("Could not securely connect to %s: %v", server, err)
+			os.Exit(1)
+		} else {
+			topContext = context.WithValue(topContext, cavalidator.CacertsValid, true)
+			systemStoreConnectionCheckRequired = false
+		}
```

Comment:
I think that this case would need some extra logic for the case where there was an error but cluster.CAStrictVerify() == false. We'd have to evaluate if we should nil out the transport since that sees re-use later on in the logic here: https://github.com/rancher/rancher/blob/ce3581538d0232a71719aac3f0ba73ae9b5f26ef/cmd/agent/main.go#L337. In the case that caused the defect, the transport would be returned but would be invalid.

Maybe something like (didn't test this):
```suggestion
                err = httpClient.Get(server)
		if err != nil {
		        if cluster.CAStrictVerify(){
		                logrus.Errorf("Could not securely connect to %s: %v", server, err)
			        os.Exit(1)
		        }
		        // systemStoreConnectionCheckRequired is still true, later logic will validate the no-transport case
		        transport = nil
		} else {
			topContext = context.WithValue(topContext, cavalidator.CacertsValid, true)
			systemStoreConnectionCheckRequired = false
		}
```

Reply:
good catch